src/components/modals/coreModal/coreModal.component.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { FC, useContext } from 'react';
+import { ModalType } from '@app/components/modals/coreModal/coreModal.interface';
+import { ModalContext } from '@app/components/modals/coreModal/coreModal.context';
+import { useStyle } from '@app/components/modals/coreModal/coreModal.component.style';
+import SuccessModal from '@app/components/modals/successModal/successModal.component';
+import DeleteModal from '@app/components/modals/deleteMovieModal/deleteMovieModal.component';
+import MovieDetailsModal from '@app/components/modals/movieDetailsModal/movieDetailsModal.component';
+
+const CoreModal: FC<Record<string, unknown>> = () => {
+  const s = useStyle();
+  const { chosenModal, setChosenModal } = useContext(ModalContext);
+  let modal: JSX.Element;
+
+  switch (chosenModal.type) {
+    case ModalType.Delete:
+      modal = (
+        <DeleteModal
+          onConfirmClick={() => setChosenModal({
+            actionType: ModalType.Delete,
+            movie: chosenModal.movie,
+            type: null,
+          })}
+          onCancelClick={() => setChosenModal({ type: null })}
+        />
+      );
+      break;
+    case ModalType.Success:
+      modal = (
+        <SuccessModal
+          onCancelClick={() => setChosenModal({ type: null })}
+        />
+      );
+      break;
+    default:
+      modal = (
+        <MovieDetailsModal
+          onConfirmClick={(movie) => setChosenModal({
+            movie,
+            type: chosenModal.type === ModalType.Add ? ModalType.Success : null,
+            actionType: chosenModal.type,
+          })}
+          onCancelClick={() => setChosenModal({ type: null })}
+          modalDetails={chosenModal}
+        />
+      );
+      break;
+  }
+
+  return (
+    <section className={s.modalBackground}>{modal}</section>
```

Comment:
Closing functionality was added.
As for extracting some elements into the common component I see that 2 buttons are presented in 1 component, 1 button in another, and the last modal has no buttons at all, I doubt that there is a big sense to handle this situation in a single place, and extracting only the close button looks like kind of microoptimization.
If you don't mind I'd prefer to leave it as it is.

Reply:
well, on projects I've worked on those elements are usually presented in a single modal component. Like closing button, title, footer buttons, like primary and secondary. So I noticed these elements in the app design and suggest these changes
Of course, you can leave it, there would be not much duplication

Reply:
Yeah, agree, all the modals in general have a common structure and shared parts. I don't remember why I chose that way, but since it's for the learning purpose only and we aren't building a production-ready application, I hope this might be some kind of excuse :)